rq/exceptions.py

```diff
@@ -30,3 +30,7 @@ def __init__(self, msg, extra_info):
 
 class TimeoutFormatError(Exception):
     pass
+
+
+class JobExpiryError(Exception):
```

Comment:
I'm debating whether `Job Expiry` is descriptive for this. This error happens because a job is left on the started job registry past it's supposed timeout, right? Any other ideas?

Reply:
Kind of, yeah. And if there are no retries left.
Maybe the failure callback should be called even if there are retries left? As happens on real job failure and no on registry cleanup?

Re:naming, I can call it I UnfinishedJobError or AbandonedJobError, I like the 2nd one more

Reply:
Yeah, I'm leaning towards `JobTimeoutError` (the execution has timed out) or `AbandonedJobError`. What do you think? This error needs to be documented and explained.

Reply:
Renamed, added to docs and calling the failure_callback always - whether there is a retry or not